tests/cases/compiler/classVarianceCircularity.ts

```diff
@@ -0,0 +1,16 @@
+// @strict: true
+
+// Issue #52813

```

Comment:
This is an example where this PR's approach doesn't solve the circularity problem.

Reply:
I assume you mean the other test with the function calls rather than expressions, since that's the one that still has an error. :) If you want, you should file an issue about the other form, so the issue can still get backlog'd, if you want to consider looking into a broader fix in the future after this PR marks the original issue as fixed.